### ISSUE TITLE
libuv writing to child stderr during spawn when some system calls fail

### ISSUE THREAD

**Issue by sam-github** [joyent/libuv#1128]
execvp(): No such file or directory, should libuv really be calling perror()?                                                                                                         It's writing the error number to the parent, and direct writes to fd 2 makes it hard in node to cleanly handle non-existence of a command, try:

```
node -e 'require("child_process").spawn("no-such-command", {stdio:"inherit"}).on("error",function(){})'
```

output is

```
execvp(): No such file or directory
```

(expected is nothing, the error is handled).

I was hoping it was stray debug code, but https://github.com/joyent/libuv/blob/master/src/unix/process.c#L362 has been there since a30e45f1, last June. The setuid and setgid just above also perror().

With this, there is no way in node to spawn, catch ENOENT, and report it without unexpected output on stderr.

/to @indutny

**Comment by indutny** [comment]
Should be fixed by https://github.com/joyent/libuv/pull/1157

**Comment by saghul** [comment]
This landed in the 0.10 branch, it will get into master next time we do a merge. Closing.
